app/controllers/passwords_controller.rb

```diff
@@ -16,17 +26,24 @@ def update
       @user.reset_api_key! if reset_params[:reset_api_key] == "true"
       sign_in @user
       redirect_to url_after_update
+      track_castle_event(Castle::ProfileUpdateSucceeded, @user)
       session[:password_reset_token] = nil
     else
       flash_failure_after_update
+      track_castle_event(Castle::ProfileUpdateFailed, @user)
       render template: "passwords/edit"
     end
+  rescue ActionController::ParameterMissing => e
```

Comment:
adding the whole rescue flow worries me a bit as it was not present before, why do we have to do this?

Reply:
@mensfeld Any parameter missing error is handled by the application controller https://github.com/castle/rubygems.org/blob/614b1b498b784a71a579e46c8e9855c724ac15b1/app/controllers/application_controller.rb#L23

I wasn't sure how to be able to know which route was causing this error from the application controller to add the castle track event there. So I overrode the default rescue in the routes that needed a castle event to be sent during this error.

Reply:
You can ignore that - that is a rare case when someone is requesting to send data without all the form input. I think we can ignore this whole case ATM. It will remove majority of the complexity in this code.